numba_dpex/core/datamodel/models.py

```diff
@@ -197,6 +230,10 @@ def _init_data_model_manager() -> datamodel.DataModelManager:
     # model manager. The dpex_data_model_manager is used by the DpexKernelTarget
     dmm.register(DpctlSyclQueue, SyclQueueModel)
 
+    # Register the DpctlSyclEvent type to SyclQueueModel in numba_dpex's data
+    # model manager. The dpex_data_model_manager is used by the DpexKernelTarget
+    dmm.register(DpctlSyclEvent, SyclEventModel)
```

Comment:
I feel adding the `SyclEventModel` to `dpex_datamode_manager` is not needed as an event object should not be used inside a kernel. Same with `SyclQueueModel`.

Reply:
I'll remove SyclEvent in this PR and SyclQueue in separate PR that will be targeted to fix it along with all other issues.